### PR TITLE
Profile, EditProfile and News implementation

### PR DESCRIPTION
**Pull request by Maksym637**
Please check my work !

### CODE REVIEW

File: src/main/java/org/ssu/edu/teachua/ui/pages/news/NewsPage.java [hunk 1]
```diff
@@ -1,10 +1,78 @@
...
+
+    private List<NewsCardComponent> newsComponents;
+
+    private List<ClubCardComponent> clubComponents;
+
+
     public NewsPage(WebDriver driver) {
         super(driver);
+        newsComponents = fillComponentsWithNews();
+        clubComponents = fillComponentsWithClubs();
```
Comment (lhalam) [inline src/main/java/org/ssu/edu/teachua/ui/pages/news/NewsPage.java:36]: remove this

File: src/main/java/org/ssu/edu/teachua/ui/base/Base.java [hunk 2]
```diff
@@ -1,19 +1,44 @@
...
+import java.time.Duration;
+import java.util.List;
 
 public abstract class Base {
     protected WebDriver driver;
+    protected WebDriverWait wait;
 
     public Base(WebDriver driver) {
         this.driver = driver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(100));
```
[the remark this answers is not in the document]
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/base/Base.java:18]: done

File: src/main/java/org/ssu/edu/teachua/ui/base/BaseComponent.java [hunk 3]
```diff
@@ -6,10 +6,16 @@
...
     public BaseComponent(WebDriver driver, WebElement node) {
         super(driver);
         componentRoot = node;
-        PageFactory.initElements(new DefaultElementLocatorFactory(componentRoot), this);
+        PageFactory.initElements(new DefaultElementLocatorFactory(componentRoot), componentRoot);
+    }
+
+    public WebElement getCurrentRoot() {
+        return componentRoot;
     }
```
[the remark this answers is not in the document]
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/base/BaseComponent.java:20]: done

File: src/main/java/org/ssu/edu/teachua/ui/components/card/ClubCardComponent.java [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
     }
+
+    public String getClubTitle() {
+        return getCurrentRoot().findElement(By.className(clubTitle)).getText();
+    }
+
+    public ViewClubPage clickDetailsButton() {
+        getCurrentRoot().findElement(By.className(detailsButton)).click();
+        return new ViewClubPage(driver);
+    }
```
[the remark this answers is not in the document]
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/components/card/ClubCardComponent.java:25]: done

File: src/main/java/org/ssu/edu/teachua/ui/components/modal/EditProfileComponent.java [hunk 5]
```diff
@@ -1,11 +1,143 @@
...
+    private WebElement confirmPasswordField;
+
+    @FindBy(how = How.XPATH, using = "//div[@class='ant-form-item-explain-error']")
+    private List<WebElement> errorMessages;
+
+    @FindBy(how = How.XPATH, using = "//button[contains(@class, 'submit-button')]")
+    private WebElement saveChangesButton;
+
+    @FindBy(how = How.XPATH, using = "(//button[@class='ant-modal-close'])[2]")
+    private WebElement closeButton;
```
[the remark this answers is not in the document]
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/components/modal/EditProfileComponent.java:53]: done

File: src/main/java/org/ssu/edu/teachua/ui/components/modal/EditProfileComponent.java [hunk 6]
```diff
@@ -1,11 +1,143 @@
...
         super(driver, node);
     }
+
+    public String getEditProfileTitle() {
+        return waitForElementToAppear(editProfileTitle).getText();
+    }
+
+    public EditProfileComponent enterNewLastName(String lastName) {
+        waitForElementToBeClickable(editLastNameField).click();
+        waitForElementToBeClickable(editLastNameField).sendKeys(
```
[the remark this answers is not in the document]
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/components/modal/EditProfileComponent.java:65]: done

File: src/main/java/org/ssu/edu/teachua/ui/pages/news/NewsPage.java [hunk 7]
```diff
@@ -1,10 +1,78 @@
...
+    private WebElement newsTitle;
+
+    @FindBy(how = How.XPATH, using = "//div[@class='sider-header']//h2")
+    private WebElement clubsTitle;
+
+    @FindBy(how = How.XPATH, using = "//div[@id='newsContainer']")
+    private List<WebElement> newsLocators;
+
+    @FindBy(how = How.XPATH, using = "//div[@class='ant-card-body']")
+    private List<WebElement> clubLocators;
```
[the remark this answers is not in the document]
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/pages/news/NewsPage.java:26]: done

File: src/main/java/org/ssu/edu/teachua/ui/pages/news/NewsPage.java (hunk 1 above)
Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/ui/pages/news/NewsPage.java:36]: done

**Review by kapustin-ilya — APPROVED** [review]

**Pull request merged**